pkg/ingester/ingester.go

```diff
@@ -2989,3 +3046,23 @@ func (i *Ingester) UserRegistryHandler(w http.ResponseWriter, r *http.Request) {
 		Timeout:            10 * time.Second,
 	}).ServeHTTP(w, r)
 }
+
+// checkReadOverloaded checks whether the ingester read path is overloaded wrt. CPU and/or memory.
+func (i *Ingester) checkReadOverloaded() error {
+	if !i.cfg.utilizationLimitsEnabled {
+		return nil
+	}
+
+	reason := i.utilizationBasedLimiter.LimitingReason()
+	if reason == "" {
+		return nil
+	}
+
```

Comment:
I would add a counter metric to track each time a request is limited. The reason is that we're return 500 error code, so there's no way to distinguish such errors from other server errors when querying the generic `cortex_request_duration_seconds` metric. If you're looking for a metric name, an option is `cortex_ingester_rate_limited_requests_total{reason="cpu|memory"}`.

Reply:
I was actually going to ask if we should add a metric.

Reply:
Done.